index.d.ts

```diff
@@ -0,0 +1,590 @@
+}
+export type ConsumerHeartbeatEvent = InstrumentationEvent<{
+    groupId: string,
+    memberId: string,
+    groupGenerationId: number,
+}>;
+export type ConsumerCommitOffsetsEvent = InstrumentationEvent<{
+    groupId: string,
+    memberId: string,
+    groupGenerationId: number,
+    topics: {
+        topic: string,
+        partitions: {
+            offset: string,
+            partition: string,
+        }[]
+    }[],
+}>;
+export interface IMemberAssignment {
+    [key: string]: number[];
+}
+export type ConsumerGroupJoinEvent = InstrumentationEvent<{
+    duration: number,
+    groupId: string,
+    isLeader: boolean,
+    leaderId: string,
+    groupProtocol: string,
+    memberId: string,
+    memberAssignment: IMemberAssignment;
+}>;
+export type ConsumerFetchEvent = InstrumentationEvent<{
+    numberOfBatches: number,
+    duration: number,
+}>;
+interface IBatchProcessEvent {
+    topic: string;
+    partition: number,
+    highWatermark: string,
+    offsetLag: string,
+    batchSize: number,
+    firstOffset: string,
+    lastOffset: string,
+}
+export type ConsumerStartBatchProcessEvent = InstrumentationEvent<IBatchProcessEvent>;
+export type ConsumerEndBatchProcessEvent = InstrumentationEvent<IBatchProcessEvent & { duration: number }>;
+export type ConsumerCrashEvent = InstrumentationEvent<{
+    error: Error,
+    groupId: string,
+}>;
+
+export type Consumer = {
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    subscribe(topic: { topic: string; fromBeginning?: boolean }): Promise<void>;
+    stop(): Promise<void>;
+    run(config?: {
+        autoCommit?: boolean;
+        autoCommitInterval?: number | null;
+        autoCommitThreshold?: number | null;
+        eachBatchAutoResolve?: boolean;
+        partitionsConsumedConcurrently?: number;
+        eachBatch?: (
+            batch: {
+                batch: Batch;
+                resolveOffset(offset: string): void;
+                heartbeat(): Promise<void>;
+                commitOffsetsIfNecessary(offsets?: Offsets): Promise<void>;
+                uncommittedOffsets(): Promise<void>;
+                isRunning(): boolean;
+            },
+        ) => Promise<void>;
+        eachMessage?: (message: { topic: string; partition: number; message: KafkaMessage }) => Promise<void>;
+    }): Promise<void>;
+    seek(topicPartition: { topic: string; partition: number; offset: string }): void;
+    describeGroup(): Promise<GroupDescription>;
+    pause(topicPartitions: TopicPartitions[]);
+    resume(topicPartitions: TopicPartitions[]);
+    on(eventName: ValueOf<ConsumerEvents>, listener: (...args: any[]) => void): void;
+    logger(): Logger;
+    events: ConsumerEvents;
+};
+
+export enum CompressionTypes {
+    None = 0,
+    GZIP = 1,
+    Snappy = 2,
+    LZ4 = 3,
+    ZSTD = 4,
+}
+
+export const CompressionCodecs: {
+    [CompressionTypes.GZIP]: () => any;
+    [CompressionTypes.Snappy]: () => never;
```

Comment:
Users of kafkajs can provide an implementation for the other codecs, how can we describe this here? (maybe any for all of theme)

Reply:
changed this to a `var` that will allow others to implement their own codes instead of a `const`